lib/src/modules/macho/mod.rs

```diff
@@ -446,6 +446,39 @@ fn import_hash(ctx: &mut ScanContext) -> Option<RuntimeString> {
     Some(RuntimeString::new(digest))
 }
 
+/// Returns an md5 hash of the symbol table in the mach-o binary
+#[module_export]
+fn sym_hash(ctx: &mut ScanContext) -> Option<RuntimeString> {
+    let macho = ctx.module_output::<Macho>()?;
+    let mut md5_hash = Md5::new();
+    let mut symtab_to_hash = &macho.symtab.entries;
+
+    // if there is not a symbtol table in the main Macho, the symbol table of the
+    // nested file should be hashed
+    if symtab_to_hash.is_empty() && !macho.file.is_empty() {
+        symtab_to_hash = &macho.file[0].symtab.entries;
+    }
+
+    // we need to check again as the nested symbol table could be empty too
+    if symtab_to_hash.is_empty() {
+        return None;
+    }
+
+    let symtab_hash_str: String = symtab_to_hash
+        .iter()
+        .filter_map(|e| {
+            Some(String::from_utf8(e.to_vec()).unwrap().trim().to_lowercase())
```

Comment:
`String::from_utf8` could fail if the symtab contains a non UTF-8 string. This is probably the case in malformed files that could make the program crash.

Reply:
Notice that you can use the `BStr` type for wrapping a `&[u8]`. This type provides `to_lowercase`, `trim` and `join`, but it works with non-UTF-8 strings.